src/contracts/pods/EigenPod.sol

```diff
@@ -222,61 +239,52 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
 
         emit ValidatorRestaked(validatorIndex);
 
+        uint64 validatorEffectiveRestakedBalanceGwei = _calculateEffectedRestakedBalanceGwei(validatorCurrentBalanceGwei);
+
+        //record validator's new restaked balance
+        validatorInfo.restakedBalanceGwei = validatorEffectiveRestakedBalanceGwei;
```

Comment:
maybe we can just combine the calculation and setting this? I think it's clearer -- the line below that reuses `validatorEffectiveRestakedBalanceGwei` now can instead just read from the field of the struct in memory

Reply:
whoops